dnf5/commands/download/download.cpp

```diff
@@ -179,6 +194,24 @@ void DownloadCommand::run() {
         return;
     }
 
+    if (source_option->get_value()) {
+        std::map<std::string, libdnf5::rpm::Package> source_pkgs;
+
+        for (auto & [nevra, pkg] : download_pkgs) {
+            libdnf5::rpm::PackageQuery pkg_query(full_pkg_query);
+            pkg_query.resolve_pkg_spec(pkg.get_source_name(), {}, true);
+            pkg_query.filter_arch({"src"});
+            pkg_query.filter_available();
```

Comment:
I think these filters could be done just once outside of the for loop.
You can create your own pre-filtered query and then just copy that instead of the `full_pkg_query`.